sdk/openai/ai-openai/review/ai-openai.api.md

```diff
@@ -0,0 +1,279 @@
+    maxTokens?: number;
+    messages: ChatMessage[];
+    model?: string;
+    n?: number;
+    presencePenalty?: number;
+    stop?: string[];
+    stream?: boolean;
+    temperature?: number;
+    topP?: number;
+    user?: string;
+}
+
+// @public
+export interface ChatMessage {
+    content?: string;
+    role: ChatRole;
+}
+
+// @public
+export type ChatRole = string;
+
+// @public
+export interface Choice {
+    finishReason: CompletionsFinishReason | null;
+    index: number;
+    logprobs: CompletionsLogProbabilityModel | null;
+    text: string;
+}
+
+// @public (undocumented)
+export interface ClientOptions extends ClientOptions_2 {
+}
+
+// @public
+export interface Completions {
+    choices: Choice[];
+    created: number;
+    id: string;
+    usage: CompletionsUsage;
+}
+
+// @public (undocumented)
+export interface Completions {
+    choices: Choice[];
+    created: number;
+    id: string;
+    usage: CompletionsUsage;
+}
+
+// @public
+export type CompletionsFinishReason = string;
+
+// @public
+export interface CompletionsLogProbabilityModel {
+    textOffset: number[];
+    tokenLogprobs: (number | null)[];
+    tokens: string[];
+    topLogprobs: Record<string, number | null>[];
+}
+
+// @public
+export interface CompletionsLogProbabilityModel {
+    textOffset: number[];
+    tokenLogprobs: (number | null)[];
+    tokens: string[];
+    topLogprobs: Record<string, number | null>[];
+}
+
+// @public
+export interface CompletionsOptions {
+    bestOf?: number;
+    echo?: boolean;
+    frequencyPenalty?: number;
+    logitBias?: Record<string, number>;
+    logprobs?: number;
+    maxTokens?: number;
+    model?: string;
+    n?: number;
+    presencePenalty?: number;
+    prompt: string[];
+    stop?: string[];
+    stream?: boolean;
+    temperature?: number;
+    topP?: number;
+    user?: string;
+}
+
+// @public (undocumented)
+export interface CompletionsOptions {
+    bestOf?: number;
+    echo?: boolean;
+    frequencyPenalty?: number;
+    logitBias?: Record<string, number>;
+    logprobs?: number;
+    maxTokens?: number;
+    model?: string;
+    n?: number;
+    presencePenalty?: number;
+    prompt: string[];
+    stop?: string[];
+    stream?: boolean;
+    temperature?: number;
+    topP?: number;
+    user?: string;
+}
+
+// @public
+export interface CompletionsUsage {
+    completionTokens: number;
+    promptTokens: number;
+    totalTokens: number;
+}
+
+// @public
+export interface EmbeddingItem {
+    embedding: number[];
+    index: number;
+}
+
+// @public
+export interface Embeddings {
+    data: EmbeddingItem[];
+    usage: EmbeddingsUsage;
+}
+
+// @public (undocumented)
+export interface Embeddings {
+    data: EmbeddingItem[];
+    usage: EmbeddingsUsage;
+}
+
+// @public
+export interface EmbeddingsOptions {
+    input: string | string[];
+    model?: string;
+    user?: string;
+}
+
+// @public (undocumented)
+export interface EmbeddingsOptions {
+    input: string | string[];
```

Comment:
@joheredi is this misgeneration? I didn't expect `input` to be an option when it is required.

Reply:
yes indeed, let me check what's going on

Reply:
Okay so it looks like these are leaking from the RLC, but generation is correct

Reply:
Found the issue. These models shouldn't be generated as we extract optional and required and place in options bag or as positional parameters. Fixed in the codegen https://bit.ly/3IMT45N this build of the code gen should have the fix

Reply:
Awesome, I regenerated using this updated version and pushed the updates in https://github.com/Azure/azure-sdk-for-js/pull/25921/commits/d1b966a33bc3ab4c686d150d45087fc23a815506. They look good, thanks!